encord/client.py

```diff
@@ -504,10 +506,32 @@ def add_private_data_to_dataset(
         else:
             raise ValueError(f"Type [{type(private_files)}] of argument private_files is not supported")
 
-        payload = {"files": files, "integration_id": integration_id, "ignore_errors": ignore_errors}
-        response = self._querier.basic_setter(DatasetData, self._config.resource_id, payload=payload)
+        process_hash = self._querier.basic_setter(
+            DatasetDataLongPolling,
+            self._config.resource_id,
+            payload={
+                "files": files,
+                "integration_id": integration_id,
+                "ignore_errors": ignore_errors,
+            },
+        )["process_hash"]
+
+        for _ in range(10):
+            polling_response = self._querier.basic_getter(
```

Comment:
This should happen in a try-catch that keeps the loop alive if the HTTP request failed with a code that indicates a temporary error (and also does a small back-off, like 3 sec or smth). Probably with an error counter and a limit.

The list of "tolerable" codes is, probably, `{502, 503, 504, 429}`?

So, we probably want to somehow have a somewhat different logic in the "retry wrapper" here.

Reply:
> This should happen in a try-catch that keeps the loop alive if the HTTP request failed with a code that indicates a temporary error (and also does a small back-off, like 3 sec or smth). Probably with an error counter and a limit.
> 
> The list of "tolerable" codes is, probably, `{502, 503, 504, 429}`?
> 
> So, we probably want to somehow have a somewhat different logic in the "retry wrapper" here.

@alexey-cord-tech @denis-cord I need more guidance how to realise this concept given that http call to SDK is hidden below few layers of abstraction.

Reply:
You'd have networking errors that you can catch like errors of the `requests` module.

Reply:
`except requests.exceptions.RequestException as e:`
@denis-cord @alexey-cord-tech 
Do we want something more fancy?